### PR TITLE
feat(pagination, split-button, dropdown, date-picker) action-group): add setFocus method

### PR DESCRIPTION
**Pull request by benelan**
**Related Issue:** #5147

## Summary

Adds the public `setFocus` method to the `delegatesFocus` components that didn't have it. See https://github.com/Esri/calcite-components/issues/5147#issuecomment-1355194965

### CODE REVIEW

**Comment by macandcheese** [comment]
@geospatialem lets make sure to add the Methods readme component section on the docs site for these when this lands for those components without it currently.

**Comment by benelan** [comment]
> @geospatialem lets make sure to add the Methods readme component section on the docs site for these when this lands for those components without it currently.

Add/remove sections automatically when stuff changes in `docs-json`?

File: src/tests/commonTests.ts [hunk 1]
```diff
@@ -213,14 +213,14 @@ export async function focusable(componentTagOrHTML: TagOrHTML, options?: Focusab
   const tag = getTag(componentTagOrHTML);
   const element = await page.find(tag);
   const focusTargetSelector = options?.focusTargetSelector || tag;
-
+  await element.callMethod("componentOnReady");
```
Comment (jcfranco) [inline src/tests/commonTests.ts:216]: Is this needed? `page.waitForChanges()` [already does this](https://github.com/ionic-team/stencil/blob/16b8ea4dabb22024872a38bc58ba1dcf1c7cc25b/src/testing/puppeteer/puppeteer-page.ts#L337).

**Review by jcfranco — APPROVED** [review]
Awesome!

File: src/components/date-picker/date-picker.tsx [hunk 2]
```diff
@@ -188,6 +195,19 @@ export class DatePicker implements LocalizedComponent, T9nComponent {
...
+  //--------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  //--------------------------------------------------------------------------
+
+  /** Sets focus on the component's first focusable element. */
+  @Method()
+  async setFocus(): Promise<void> {
+    await componentLoaded(this);
```
[the remark this answers is not in the document]
Comment (benelan) [reply src/components/date-picker/date-picker.tsx:207]: Yeah that could be nice. Matt started a PR (#6156) to try to cut down the `setFocus` boilerplate but from our discussion it sounds like we can't use `componentOnReady()` internally. So your suggestion is a good plan B

**Pull request merged**